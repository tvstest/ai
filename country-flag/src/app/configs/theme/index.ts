import { createTheme } from '@mui/material'

const appTheme = createTheme({
  typography: {
    fontFamily: 'Montserrat',
  },
})

export default appTheme
