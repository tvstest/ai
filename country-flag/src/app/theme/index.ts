import { createTheme } from "@material-ui/core";

const appTheme = createTheme({
  typography: {
    fontFamily: "Nunito",
  },
});

export default appTheme;
