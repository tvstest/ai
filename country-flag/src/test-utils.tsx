import React from 'react'
import { render, RenderOptions, RenderResult } from '@testing-library/react'
import { ThemeProvider } from '@mui/material/styles'
import Navbar from 'app/components/Navbar'
import { BrowserRouter, Redirect, Route, Switch } from 'react-router-dom'
import { Routes } from 'app/utils/enums/routes'
import appTheme from 'app/configs/theme'
import CountrySearch from 'app/pages/CountrySearch'
import CountryList from 'app/pages/CountryList'

const AppComponent: React.FC = () => {
  return (
    <ThemeProvider theme={appTheme}>
      <BrowserRouter>
        <Navbar />
        <Switch>
          <Route exact path={Routes.Home} component={CountrySearch} />
          <Route exact path={Routes.CountriesByName} component={CountryList} />
          <Redirect to={Routes.Home} />
        </Switch>
      </BrowserRouter>
    </ThemeProvider>
  )
}

const customRender = (
  ui: React.ReactElement,
  options?: RenderOptions
): RenderResult => render(ui, { wrapper: AppComponent, ...options })

export * from '@testing-library/react'
export { customRender as render }
