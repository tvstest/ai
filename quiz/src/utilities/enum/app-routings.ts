export enum AppRoutings {
  Registration = '/',
  Quiz = '/quiz',
  Report = '/report',
}
