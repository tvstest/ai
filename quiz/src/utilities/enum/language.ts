export enum Language {
  English,
  Hindi,
}
