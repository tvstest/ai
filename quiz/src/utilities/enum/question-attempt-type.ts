export enum QuestionAttemptType {
  Answered,
  NotAnswered,
}

export enum QuestionCorrection {
  Correct = 'Correct',
  Incorrect = 'Incorrect',
  NotAnswered = 'Not answered',
}

export enum QuestionColor {
  SUCCESS = 'success',
  ERROR = 'error',
  WARNING = 'warning',
}
