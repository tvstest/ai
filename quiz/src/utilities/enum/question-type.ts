export enum QuestionType {
  SingleSelect,
  CorrectIncorrect,
  FillInTheBlanks,
  MatchTheFollowing,
  MultiSelect,
}
