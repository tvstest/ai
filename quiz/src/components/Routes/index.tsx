/* eslint-disable react/no-array-index-key */
import Layout from 'components/Layout'
import { lazy } from 'react'
import { Switch, Route } from 'react-router-dom'

import { AppRoutings } from 'utilities/enum/app-routings'

interface IRoute {
  path: string
  component:
    | React.LazyExoticComponent<React.FC<{}>>
    // | React.ComponentType<RouteComponentProps<{}>>
    | React.ComponentType<{}>
  isProtectedRoute: boolean
}

const routes: IRoute[] = [
  {
    path: AppRoutings.Registration,
    component: lazy(() => import('pages/Registration')),
    isProtectedRoute: false,
  },
  {
    path: AppRoutings.Quiz,
    component: lazy(() => import('pages/Quiz')),
    isProtectedRoute: false,
  },
  {
    path: AppRoutings.Report,
    component: lazy(() => import('pages/ReportCard')),
    isProtectedRoute: false,
  },
]

const Routes: React.FC = () => {
  return (
    <Switch>
      <Layout>
        {routes
          .filter((route) => !route.isProtectedRoute)
          .map(({ path, component }, key) => (
            <Route exact path={path} component={component} key={key} />
          ))}
      </Layout>
    </Switch>
  )
}

export default Routes
