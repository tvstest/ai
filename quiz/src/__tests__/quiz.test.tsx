import { fireEvent, render, waitFor } from '@testing-library/react'
import { createMemoryHistory } from 'history'
import { AppRoutings } from 'utilities/enum/app-routings'
import { Language } from 'utilities/enum/language'
import '@testing-library/jest-dom'
import CircularProgress from '@mui/material/CircularProgress'
import theme from 'configs/app-theme'
import Routes from 'components/Routes'
import { Suspense } from 'react'
import { Router } from 'react-router'
import { ThemeProvider } from '@mui/material/styles'
import FillInTheBlank from 'components/QuestionType/FillInTheBlanks'
import { act } from 'react-dom/test-utils'

test("On selecting answer it's relevant handler is called", async () => {
  const mockFn = jest.fn()
  const { getByTestId } = render(
    <FillInTheBlank userAnswer="Danial" handleAnswer={mockFn} />
  )

  const inputElement = getByTestId('fillInput')

  await act(async () => {
    fireEvent.change(inputElement, {
      target: { value: 'east' },
    })
  })

  await waitFor(async () => {
    expect(mockFn).toHaveBeenCalledTimes(1)
  })
})

test('Input component matches the snapshot', () => {
  const mockFn = jest.fn()
  const element = render(
    <FillInTheBlank userAnswer="Danial" handleAnswer={mockFn} />
  )
  expect(element).toMatchSnapshot()
})

test('After submitting form user is navigated to quiz page', async () => {
  const history = createMemoryHistory()
  history.push({
    pathname: AppRoutings.Quiz,
    state: { language: Language.English, name: 'Danial' },
  })
  const { getByText } = render(
    <Suspense fallback={<CircularProgress />}>
      <ThemeProvider theme={theme}>
        <Router history={history}>
          <Routes />
        </Router>
      </ThemeProvider>
    </Suspense>
  )
  await waitFor(async () => expect(getByText(/Hi Danial/i)).toBeInTheDocument())
})
