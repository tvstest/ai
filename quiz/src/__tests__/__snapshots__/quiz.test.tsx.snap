// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component matches the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiFormControl-root MuiTextField-root css-1qf00zc-MuiFormControl-root-MuiTextField-root"
      >
        <div
          class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            data-testid="fillInput"
            id="outlined-required"
            required=""
            type="text"
            value="Danial"
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-nnbavb"
            >
              <span
                class="notranslate"
              >
                ​
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiFormControl-root MuiTextField-root css-1qf00zc-MuiFormControl-root-MuiTextField-root"
    >
      <div
        class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-invalid="false"
          class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          data-testid="fillInput"
          id="outlined-required"
          required=""
          type="text"
          value="Danial"
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-nnbavb"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
